src/sentry/migrations/0029_discover_query_upgrade.py

```diff
@@ -0,0 +1,201 @@
+            updated_query["query"].append(u"!{}:{}".format(column, prepare_value(value)))
+    updated_query["query"] = ' '.join(updated_query["query"])
+
+    # Create the version 2 query
+    new_query = DiscoverSavedQuery.objects.create(
+        organization=saved_query.organization,
+        name=saved_query.name + name_extra,
+        query=updated_query,
+        version=2,
+    )
+
+    # Set project_ids
+    saved_query_project_ids = DiscoverSavedQueryProject.objects.filter(
+        discover_saved_query=saved_query
+    ).values_list("project", flat=True)
+
+    # This is DiscoverSavedQueryProject.set_projects
+    DiscoverSavedQueryProject.objects.filter(discover_saved_query=new_query).exclude(
+        project__in=saved_query_project_ids
+    ).delete()
+
+    existing_project_ids = DiscoverSavedQueryProject.objects.filter(
+        discover_saved_query=new_query
+    ).values_list("project", flat=True)
+
+    new_project_ids = list(set(saved_query_project_ids) - set(existing_project_ids))
+
+    DiscoverSavedQueryProject.objects.bulk_create(
+        [
+            DiscoverSavedQueryProject(project_id=project_id, discover_saved_query=new_query)
+            for project_id in new_project_ids
+        ]
+    )
+
+    return new_query
+
+def migrate_v1_queries(apps, schema_editor):
+    """
+    Creates v2 versions of existing v1 queries
+    """
+    DiscoverSavedQuery = apps.get_model("sentry", "DiscoverSavedQuery")
+    DiscoverSavedQueryProject = apps.get_model("sentry", "DiscoverSavedQueryProject")
+
+    """ Seq Scan on sentry_discoversavedquery
+        (cost=0.00..102.86 rows=1601 width=284)
+        (actual time=0.027..1.158 rows=1275 loops=1)
+        Filter: (version = 1)
+        Rows Removed by Filter: 69
+        Planning time: 0.929 ms
+        Execution time: 1.296 ms
+    """
+    queryset = DiscoverSavedQuery.objects.filter(version=1)
+
+    for query in RangeQuerySetWrapperWithProgressBar(queryset):
+        convert(DiscoverSavedQuery, DiscoverSavedQueryProject, query)
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Adding indexes to large tables. These indexes should be created concurrently,
+    #   unfortunately we can't run migrations outside of a transaction until Django
+    #   1.10. So until then these should be run manually.
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
```

Comment:
Locally this migration runs in ~30s
@wedamija do you mean like this?
```suggestion
    is_dangerous = False
    atomic = False
```

Reply:
@wmak Yep! Yeah, I think it's a good idea to mark it as non-atomic.